src/model/ModelObject.cpp

```diff
@@ -752,6 +752,51 @@ namespace model {
       return result;
     }
 
+    boost::optional<std::string> ModelObject_Impl::cadObjectId() const {
+      boost::optional<std::string> result;
+      if (hasAdditionalProperties()) {
+        AdditionalProperties additionalProperties = this->additionalProperties();
+        if (additionalProperties.hasFeature("CADObjectId")) {
+          result = additionalProperties.getFeatureAsString("CADObjectId");
+        }
+      }
+      return result;
+    }
+
+    bool ModelObject_Impl::setCADObjectId(std::string cadObjectId) {
```

Comment:
Do not make a copy of std::string here. Use a const ref
```suggestion
    bool ModelObject_Impl::setCADObjectId(const std::string& cadObjectId) {
```